src/oauth2c.erl

```diff
@@ -353,6 +379,52 @@ add_auth_header(Headers, #client{access_token = AccessToken}) ->
   AH = {<<"Authorization">>, <<"token ", AccessToken/binary>>},
   [AH | proplists:delete(<<"Authorization">>, Headers)].
 
+get_cached_token(Key) ->
+  case whereis(oauth2c_token_cache) of
+    undefined ->
+      cache_server_not_started;
+    _ ->
+      oauth2c_token_cache:get(Key)
+  end.
+
+create_token_key(Type, Url, undefined) ->
+  <<Type/binary, Url/binary>>;
+create_token_key(Type, Url, Scope) ->
+  <<Type/binary, Url/binary, Scope/binary>>.
```

Comment:
This is insufficient. This way tokens for different clients will be cached under the same key which is incorrect.

I suggest hashing the full client record. Maybe `erlang:phash2` is good enough.

Reply:
I've switched to hashing the entire client record with phash2 as you suggested.

Reply:
Nice!